velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -901,6 +903,86 @@ TpchPlan TpchQueryBuilder::getQ19Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ22Plan() const {
+  std::vector<std::string> ordersColumns = {"o_custkey"};
+  std::vector<std::string> customerColumns_1 = {"c_acctbal", "c_phone"};
+  std::vector<std::string> customerColumns_2 = {
+      "c_custkey", "c_acctbal", "c_phone"};
+
+  const auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders);
+  const auto customerSelectedRowType_1 =
+      getRowType(kCustomer, customerColumns_1);
+  const auto& customerFileColumns_1 = getFileColumnNames(kCustomer);
+  const auto customerSelectedRowType_2 =
+      getRowType(kCustomer, customerColumns_2);
+  const auto& customerFileColumns_2 = getFileColumnNames(kCustomer);
+
+  const std::string phoneFilter =
+      "substr(c_phone, 1, 2) IN ('13', '31', '23', '29', '30', '18', '17')";
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId customerScanNodeId_1;
+  core::PlanNodeId customerScanNodeId_2;
+  core::PlanNodeId ordersScanNodeId;
+
+  auto orders =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kOrders, ordersSelectedRowType, ordersFileColumns)
+          .capturePlanNodeId(ordersScanNodeId)
+          .planNode();
+
+  auto customer = PlanBuilder(planNodeIdGenerator, pool_.get())
+                      .tableScan(
+                          kCustomer,
+                          customerSelectedRowType_1,
+                          customerFileColumns_1,
+                          {"c_acctbal > 0.0"},
+                          phoneFilter)
+                      .capturePlanNodeId(customerScanNodeId_1)
+                      .partialAggregation({}, {"avg(c_acctbal) as avg_acctbal"})
+                      .localPartition({})
+                      .finalAggregation()
+                      .project({"avg_acctbal"})
+                      .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator, pool_.get())
+          .tableScan(
+              kCustomer,
+              customerSelectedRowType_2,
+              customerFileColumns_2,
+              {},
+              phoneFilter)
+          .capturePlanNodeId(customerScanNodeId_2)
+          .crossJoin(
+              customer, {"c_acctbal", "avg_acctbal", "c_custkey", "c_phone"})
+          .filter("c_acctbal > avg_acctbal")
+          .hashJoin(
+              {"c_custkey"},
+              {"o_custkey"},
+              orders,
+              "",
+              {"c_acctbal", "c_phone"},
+              core::JoinType::kAnti)
+          .project({"substr(c_phone, 1, 2) AS cntrycode", "c_acctbal"})
```

Comment:
Just wanted to double check that "substr(c_phone, 1, 2)" is not abstracted as a Common Sub-expression in the Presto plan. Please can you confirm.

Reply:
Presto abstracts this using substr variable after the join on custkey as follows:
                     substr := substr(phone, BIGINT'1', BIGINT'2') (11:9)                                                                                       >
                 - FilterProject[filterPredicate = not(COALESCE(subquerytrue, BOOLEAN'false')), projectLocality = LOCAL] => [phone:varchar(15), acctbal:double] >
                         CPU: 5.00ms (0.16%), Scheduled: 23.00ms (0.16%), Output: 73 rows (2.07kB)                                                              >
                         Input avg.: 11.88 rows, Input std.dev.: 63.78%                                                                                         >
                     - LeftJoin[("custkey" = "custkey_13")][$hashvalue_76, $hashvalue_81] => [phone:varchar(15), acctbal:double, subquerytrue:boolean]          >
                     
Apart from the use of antijoin, the current plan appears to be identical to Presto plan in terms of the subexpression calculation. Could you please share if any changes are needed?